### PR TITLE
Unique nulljunk refactor

### PR DESCRIPTION
**Pull request by talekhinezh**
* Refactor of IdParticle and NullJunk into UniqueParticle.
* Translation from UniqueProperty to UniqueParticle

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -86,15 +86,19 @@ selected identities can read the data.
 To store the encrypted string `Hello` which only I can read into an account:
 ```
 ECPublicKey myPublicKey = api.getMyPublicKey();
-Data data = new DataBuilder().bytes("Hello".getBytes()).addReader(myPublicKey)
-			.build();
+Data data = new DataBuilder()
+    .bytes("Hello".getBytes())
```
Comment (mpsandiford) [inline README.md]: Really should include a character set here, aka `getBytes(StandardCharsets.UTF_8)` or whatever is appropriate.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java [hunk 2]
```diff
@@ -176,35 +185,90 @@ public Result storeData(Data data, RadixAddress address0, RadixAddress address1)
...
 
-		return new Result(updates);
+	public Result transferTokens(
+		RadixAddress from,
+		RadixAddress to,
+		Amount amount,
+		@Nullable Data attachment,
+		@Nullable byte[] unique // TODO: make unique immutable
+	) {
+		Objects.requireNonNull(from);
```
Comment (mpsandiford) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: Use of @Nullable and testing prerequisites is good here.

File: radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java [hunk 3]
```diff
@@ -176,35 +185,90 @@ public Result storeData(Data data, RadixAddress address0, RadixAddress address1)
 			.share();
 	}
 
+	/**
+	 * Sends an amount of a token to an address
```
Comment (mpsandiford) [inline radixdlt-java/src/main/java/com/radixdlt/client/application/RadixApplicationAPI.java]: Muchos kudos for adding some javadoc :)

File: README.md (hunk 1 above)
Comment (talekhinezh) [reply README.md]: Cool, will fix on master branch.

**Pull request merged**